Jenkinsfile

```diff
@@ -1,14 +1,41 @@
+def nexusRepoDetails = getNexusRepoDetails()
+def mavenSettingsId = getMavenSettingsId()
+def mavenAdditionalSettings = getMavenAdditionalSettings()
+def enceladusSlaveLabel = getEnceladusSlaveLabel()
+
+
 pipeline {
-     agent {
-         docker {
-             image 'maven:3-alpine'
-         }
-     }
-     stages {
-         stage('Build') {
-             steps {
-                 sh 'mvn -B -DskipTests clean install'
-             }
-         }
-     }
- }
+    agent {
+        label "${enceladusSlaveLabel}"
+    }
+    tools { 
+        jdk 'openjdk-1.8.0'
+        maven 'Maven-3.6.0' 
+        git 'git-latest'
+    }
+    options { 
+        buildDiscarder(logRotator(numToKeepStr: '20'))
+        timestamps()
+    }
+    stages {
+        stage ('Prepare') {
+            steps {
+                enceladusCreateConfigurationFiles()
+            }
+        }
+        stage ('Build') {
+            steps {
+                configFileProvider([configFile(fileId: "${mavenSettingsId}", variable: 'MAVEN_SETTINGS_XML')]) {
+                    sh "mvn -s $MAVEN_SETTINGS_XML clean package ${mavenAdditionalSettings}"
+                }
+            }
+        }
+        stage ('Deploy') {
```

Comment:
Sounds good.
I've updated our Jenkinsfile to deploy to the snapshot repo only if the branch is develop and release only if branch is master.

The first step is clean package which is all that pull requests will run.
This will verify for each pull request that the modules build successfully and pass all tests.
Once they get merged to master, they will be deployed to the internal repos.

This can be all expanded on further without too much trouble!